astropy/io/fits/tests/test_connect.py

```diff
@@ -24,8 +24,8 @@
 from astropy.table import Column, MaskedColumn, QTable, Table
 from astropy.table.table_helpers import simple_table
 from astropy.time import Time
+from astropy.units import UnitScaleError
 from astropy.units import allclose as quantity_allclose
-from astropy.units.format.fits import UnitScaleError
```

Comment:
Unrelated to your PR, but how is `UnitScaleError` even available in `units.format.fits`? It is only used there via `core.UnitScaleError`... And there are no `*` imports. Yet,
```
import astropy.units.format.fits
dir(astropy.units.format.fits)
['FITS',
 'TYPE_CHECKING',
 'UnitScaleError',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__spec__',
 'annotations',
 'classproperty',
 'core',
 'generic',
 'np',
 'utils']
```

Reply:
OK, it is from here: https://github.com/astropy/astropy/blob/9e8806701929ad6605cb4c8f30c57229c8142eac/astropy/units/core.py#L2716-L2718

Can we just remove those lines? (Then even more it is for 7.0). I'm fine doing that without deprecation warning...

Reply:
What I really wanted to do was to move the error and warning definitions from `astropy/units/core.py` to a separate `astropy/units/errors.py` file so that other modules in `units` could import them without having to import all of `core.py`. Such a change would help avoid import loops. But when I was making that change I quickly noticed that I had copied over five classes from `core.py`, but only four entries for `__all__`, which is how I found this bug. I then decided to open this pull request first because it makes user-visible changes that deserve a change log entry. What you are suggesting here sounds like it belongs in the other pull request, which should be user-invisible refactoring.

Reply:
Happy to deal with that separately, indeed.